examples/fungible-vault/src/contract.rs

```diff
@@ -0,0 +1,109 @@
+//! Tokenized Vault Example Contract.
+
+use soroban_sdk::{contract, contractimpl, Address, Env, String};
+use stellar_macros::default_impl;
+use stellar_tokens::fungible::{
+    vault::{FungibleVault, Vault},
+    Base, FungibleToken,
+};
+
+#[contract]
+pub struct ExampleContract;
+
+#[contractimpl]
+impl ExampleContract {
+    pub fn __constructor(e: &Env, asset: Address, decimals_offset: u32) {
+        // Asset and decimal offset should be configured once during initialization.
+        Vault::set_asset(e, asset);
+        Vault::set_decimals_offset(e, decimals_offset);
+        // Vault overrides the decimals function by default.
+        // Decimal offset must be set prior to metadata initialization.
+        Base::set_metadata(
+            e,
+            Self::decimals(e),
+            String::from_str(e, "Vault Token"),
+            String::from_str(e, "VLT"),
+        );
+    }
+}
+
+#[default_impl]
+#[contractimpl]
+impl FungibleToken for ExampleContract {
+    type ContractType = Vault;
+
+    // Allows override of decimals and other base functions.
+
+    fn decimals(e: &Env) -> u32 {
+        Vault::decimals(e)
+    }
+}
+
+#[contractimpl]
```

Comment:
`#[default_impl]` support for this would be good. I can push that change as well, this is just a reminder

Reply:
Sounds good! 

> it's great to collaborate with you both!

Likewise 🥂 

Thanks a lot again for all this 🚀